### PR TITLE
Fixing big integer issue in HypreParMatrixFromBlocks.

### PR DESCRIPTION
**Pull request by dylan-copeland**
Fixes an issue with recent versions of hyper.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1521](https://github.com/mfem/mfem/pull/1521) | @dylan-copeland | @tzanio | @v-dobrev + @mlstowell | 06/07/20 | 06/26/20 | 07/04/20 | |

### CODE REVIEW

File: linalg/hypre.cpp [hunk 1]
```diff
@@ -1856,6 +1857,11 @@ HypreParMatrix * HypreParMatrixFromBlocks(Array2D<HypreParMatrix*> &blocks,
                                                (*(blocks(i, j)));
                MFEM_ASSERT(parcsr_op != NULL, "const_cast failed");
                csr_blocks(i, j) = hypre_MergeDiagAndOffd(parcsr_op);
+#if MFEM_HYPRE_VERSION >= 21600
+               MFEM_VERIFY(csr_blocks(i, j)->num_rows < INT_MAX,"Number of "
+                           "local rows is too large to store as an integer.");
```
Comment (v-dobrev) [inline linalg/hypre.cpp:1862]: 1. Why not use `<=` here?
2. The `J` array stores column indices, shouldn't the check be for `num_cols`?

Comment (dylan-copeland) [reply linalg/hypre.cpp:1862]: Good points. I copied this from `superlu.cpp` without thinking about it. Should I make the same changes to that file?

Comment (v-dobrev) [reply linalg/hypre.cpp:1862]: I think so. @mlstowell, what do you think?

Comment (v-dobrev) [reply linalg/hypre.cpp:1862]: Another thing: when `HYPRE_Int` is 64bit, the check with `INT_MAX` is wrong -- there is no need to check anything.

Comment (mlstowell) [reply linalg/hypre.cpp:1862]: When @jandrej and I discussed this in his SuperLU PR #988 we were talking about checking `num_cols`.  I'm not sure why the test uses `num_rows`.  I must have written the wrong thing.

Comment (mlstowell) [reply linalg/hypre.cpp:1862]: In the SuperLU wrapper we needed an `int*` point to `j` to pass into a SuperLU matrix data structure so we had to first copy the 64bit `HYPRE_Ints` to `ints`.  In this case I don't think the copy is necessary.  You need to check whether HYPRE is storing the column indices in `hypre_CSRMatrix::j` or `hypre_CSRMatrix::BigJ` and read them from the correct array.

Comment (dylan-copeland) [reply linalg/hypre.cpp:1862]: @mlstowell In seq_mv.h it looks like there are arrays `j` and `big_j` in `hypre_CSRMatrix`. Does that mean one is `NULL` and the other is not, in every case? If so, I can just check for which one is not `NULL`.

**Comment by v-dobrev** [comment]
LGTM

**Comment by v-dobrev** [comment]
LGTM

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**